adapters/adserver_adtarget/adserver_adtarget.go

```diff
@@ -0,0 +1,190 @@
+		Message: "request.Site or request.App are not provided",
+	}
+}
+
+func makeRequestHeaders(openRTBRequest *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+
+	if openRTBRequest.Device != nil {
+		if len(openRTBRequest.Device.UA) > 0 {
+			headers.Add("User-Agent", openRTBRequest.Device.UA)
+		}
+
+		if len(openRTBRequest.Device.IPv6) > 0 {
+			headers.Add("X-Forwarded-For", openRTBRequest.Device.IPv6)
+		}
+
+		if len(openRTBRequest.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", openRTBRequest.Device.IP)
+		}
+	}
+	return headers
+}
+
+func (v *adapter) checkResponseStatusCodes(response *adapters.ResponseData) error {
+	if response.StatusCode == http.StatusBadRequest {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: [ %d ]", response.StatusCode),
+		}
+	}
+
+	if response.StatusCode == http.StatusServiceUnavailable {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Something went wrong, please contact your Account Manager. Status Code: [ %d ] ", response.StatusCode),
+		}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: [ %d ]. Run with request.debug = 1 for more info", response.StatusCode),
+		}
+	}
+
+	return nil
+}
+
+func (v *adapter) MakeBids(
+	openRTBRequest *openrtb2.BidRequest,
+	requestToBidder *adapters.RequestData,
+	bidderRawResponse *adapters.ResponseData,
+) (
+	*adapters.BidderResponse,
+	[]error,
+) {
+	if bidderRawResponse.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	httpStatusError := v.checkResponseStatusCodes(bidderRawResponse)
+	if httpStatusError != nil {
+		return nil, []error{httpStatusError}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(bidderRawResponse.Body, &response); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Bad Server Response",
+		}}
+	}
+
+	if len(response.SeatBid) == 0 {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Empty SeatBid array",
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(openRTBRequest.Imp))
+	bidResponse.Currency = response.Cur
+	for _, seatBid := range response.SeatBid {
+		for i, bid := range seatBid.Bid {
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: getMediaTypeForImp(bid.ImpID, openRTBRequest.Imp),
+			}
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, nil
+}
+
+func getMediaTypeForImp(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			} else if imp.Native != nil {
+				mediaType = openrtb_ext.BidTypeNative
+			}
+			return mediaType
```

Comment:
I might be confusing myself, but here it looks like you support `Video` and `Native` media types in addition to `Banner`. But in your docs PR and your `adserver_adtarget.yaml` file, it looks like you only support Banner. If you support video and native as well, could you update the docs PR and .yaml file? If not, could you explain what's being done here?

Reply:
We want to start with a banner, but we added the logic beforehand for the video and native. We remove "video" and "native" support if this logic doesn't work.